tests/test_stan_sts.py

```diff
@@ -62,3 +66,54 @@ def test_stan_statefulset_with_metrics_and_resources(self, kube_version):
         c_by_name = {c["name"]: c for c in containers}
         assert c_by_name["stan"]["resources"]["requests"]["cpu"] == "123m"
         assert c_by_name["metrics"]["resources"]["requests"]["cpu"] == "234m"
+
+    def test_stan_statefulset_with_affinity_and_tolerations(self, kube_version):
+        """Test that stan statefulset renders proper nodeSelector, affinity, and tolerations"""
+        docs = render_chart(
+            kube_version=kube_version,
+            show_only=["charts/stan/templates/statefulset.yaml"],
+            values={
+                "stan": {
+                    "nodeSelector": {"role": "astro"},
+                    "affinity": {
+                        "nodeAffinity": {
+                            "requiredDuringSchedulingIgnoredDuringExecution": {
+                                "nodeSelectorTerms": [
+                                    {
+                                        "matchExpressions": [
+                                            {
+                                                "key": "astronomer.io/multi-tenant",
+                                                "operator": "In",
+                                                "values": ["false"],
+                                            }
+                                        ]
+                                    }
+                                ]
+                            }
+                        }
+                    },
+                    "tolerations": [
+                        {
+                            "effect": "NoSchedule",
+                            "key": "astronomer",
+                            "operator": "Exists",
+                        }
+                    ],
+                },
+            },
+        )
+
+        assert len(docs) == 1
+        spec = docs[0]["spec"]["template"]["spec"]
+        assert spec["nodeSelector"] != {}
+        assert spec["nodeSelector"]["role"] == "astro"
+        assert spec["affinity"] != {}
+        assert (
+            len(
+                spec["affinity"]["nodeAffinity"][
+                    "requiredDuringSchedulingIgnoredDuringExecution"
+                ]["nodeSelectorTerms"]
+            )
+            == 1
+        )
+        assert len(spec["tolerations"]) > 0
```

Comment:
Same comment as above, maybe we want to be more specific on the assertion here to test the right behavior.